src/libsyntax/ext/tt/macro_parser.rs

```diff
@@ -825,7 +825,7 @@ fn may_begin_with(token: &Token, name: Name) -> bool {
     }
 
     match name {
-        sym::expr => token.can_begin_expr(),
+        sym::expr => token.can_begin_expr() && !token.is_keyword(kw::Let),
```

Comment:
Could you add a comment saying that the exception is here for backward compatibility.

Reply:
Done.